tests/test_integration/test_dace_parsing.py

```diff
@@ -49,93 +54,119 @@ def tuple_st(min_value, max_value):
     )
 
 
-def test_basic():
-    @gtscript.stencil(backend="dace:cpu")
+@pytest.mark.parametrize(
+    "backend",
+    ["dace:cpu", pytest.param("dace:gpu", marks=[pytest.mark.requires_gpu])],
+)
+def test_basic(backend):
+    @gtscript.stencil(backend=backend)
     def defn(outp: gtscript.Field[np.float64], par: np.float64):
         with computation(PARALLEL), interval(...):
             outp = par  # noqa F841: local variable 'outp' is assigned to but never used
 
     outp = gt_storage.zeros(
-        dtype=np.float64, shape=(10, 10, 10), default_origin=(0, 0, 0), backend="dace:cpu"
+        dtype=np.float64, shape=(10, 10, 10), default_origin=(0, 0, 0), backend=backend
     )
 
     inp = 7.0
 
-    @dace.function
-    def call_stencil_object():
-        defn(outp, par=inp)
+    outp.host_to_device()
 
-    call_stencil_object()
+    @dace.program(device=dace.DeviceType.GPU if "gpu" in backend else dace.DeviceType.CPU)
+    def call_stencil_object(locoutp, locinp):
+        defn(locoutp, par=locinp)
 
+    call_stencil_object(locoutp=outp, locinp=inp)
+    if hasattr(outp, "_set_device_modified"):
+        outp._set_device_modified()
+    outp.device_to_host()
```

Comment:
I think we should be able to use `outp.device_to_host(force=True)` instead of using the protected api method `_set_device_modified`.